File: benchmarks/workloads/filter_map_reduce/javascript/bench.js
function* rangeGen(n) {
    for (let i = 0n; i < n; i++) yield i;
}

const MASK = (1n << 64n) - 1n;
const n = BigInt(process.argv[2]);
let acc = 0n;
for (const v of rangeGen(n)) {
    if ((v & 1n) === 0n) acc = (acc + v * v) & MASK;
}
console.log(`result = ${acc}`);

File: benchmarks/workloads/filter_map_reduce/python/bench.py
import sys

def range_gen(n):
    i = 0
    while i < n:
        yield i
        i += 1

def main():
    n = int(sys.argv[1])
    MASK = (1 << 64) - 1
    acc = 0
    for v in range_gen(n):
        if v % 2 == 0:
            acc = (acc + v * v) & MASK
    print(f"result = {acc}")

if __name__ == "__main__":
    main()
